### PR TITLE
Removing call to parent class constructor from WPcom_JS_Concat's one

### PR DESCRIPTION
**Pull request by david-binda**
Since the `WP_Scripts::__construct` has either already been called outside the WPcom_JS_Concat class or is called from within the `WPcom_JS_Concat::__construct` during the initialisation of the `old_scripts` variable, we should not call the parent class consturctor again, as [it fires](https://github.com/WordPress/WordPress/blob/master/wp-includes/class.wp-scripts.php#L143) the `wp_default_scripts` action again.

The `wp_default_scripts` action is designed for 2 runs, before and after the `init` action. [If it's run after the `init` action, localisations are being populated](https://github.com/WordPress/WordPress/blob/493f76a3d2ef8c030ab5dcd4333f9a401208f534/wp-includes/script-loader.php#L89) and calling that action for the second time after the `init` action leads to doubling the localisation strings.

### CODE REVIEW

**Comment by joshbetz** [comment]
Do we need to fix this in cssconcat as well?

**Comment by david-binda** [comment]
Good question. Let me check that.

**Comment by david-binda** [comment]
@joshbetz despite there are no related bugs to that call, it looks like it might be better to remove that call as the `WP_Styles` class is not written as singleton and it triggers similar action and I don't think that the action should be triggered multiple times.

File: jsconcat.php [hunk 1]
```diff
@@ -31,8 +31,6 @@ function __construct( $scripts ) {
 			}
 			unset( $this->$key );
 		}
-
-		parent::__construct();
```
Comment (david-binda) [inline jsconcat.php]: That definitely would do the trick, but the parent constructor would [still register an extra action](https://github.com/WordPress/WordPress/blob/dd6da701b286579819cd6aa518aa2d7018efd759/wp-includes/class.wp-scripts.php#L145) which does not feel right, especially as that action would trigger a method which does nothing.

> That feels like the more "appropriate" approach than not calling the parent constructor.

I would usually agree, but the way the `WPcom_JS_Concat` class is implemented makes me think otherwise. It is designed to replace an, [usually](https://github.com/Automattic/nginx-http-concat/commit/dac2aa0ad3b65c041e7f393a4c63e447574c143a), already existing object and override some of it's methods, while keeping the [original object's properties in place](https://github.com/Automattic/nginx-http-concat/blob/master/jsconcat.php#L25,L34).

In an ideal world, the `WPcom_JS_Concat` class would be used for initialising the global `$wp_scripts` variable, triggering all the actions and getting all the properties set directly.

I really wonder whether there was a good reason for hooking the `js_concat_init` fuction to `init` at priority 10, instead of trying to be called before the global object is created. Brief testing shows that priority set to 0 would do the trick, but it might not be as easy as that.

So, in summary, if we believe that the parent's constructor should be called, then we should revisit the whole implementation, which is not something I wanted to perform in this PR, but am happy to dive into it.

Comment (joshbetz) [inline jsconcat.php]: The parent constructor is called here? https://github.com/Automattic/nginx-http-concat/blob/master/jsconcat.php#L19-L23

Either we pass in a WP_Scripts object (so it has already been called) or we instantiate a new WP_Scripts object, which is the same as calling the constructor.

Comment (david-binda) [inline jsconcat.php]: That's exactly what I tried to describe in my comment :)

The question whether the whole implementation is correct is still open, imho. But again, I don't feel like it should be a subject of this bug fixing PR, but am open to discussion.

Comment (mjangda) [inline jsconcat.php]: That makes sense, thanks @david-binda!
